### PR TITLE
Feature/multiple addresses

### PR DESCRIPTION
**Pull request by mr-rooftop**
new day, new PR

### CODE REVIEW

File: thornode_bot.py [hunk 1]
```diff
@@ -185,68 +202,78 @@ def cancel(update, context):
...
-        user_data['status'] = node['status']
-        user_data['bond'] = node['bond']
-        user_data['slash_points'] = node['slash_points']
-
-        # Send message
-        context.bot.send_message(chat_id, text)
-
-    if len(changed_fields) > 0:
-        show_action_buttons(context, chat_id=chat_id)
+    # flag to show home buttons or not
```
Comment (rekt0x) [inline thornode_bot.py:221]: Start comments with an uppercase letter, to keep consistency. Only mentioned it here, but there are a few occurences.

File: thornode_bot.py [hunk 2]
```diff
@@ -185,68 +202,78 @@ def cancel(update, context):
...
+    message_sent = False
+
+    # list to delete entries after loop
+    delete_addresses = []
+    
+    # iterate through all keys
+    for address in user_data.keys():
+        # Filter out the thornode addresses
+        if "thor" in address:
```
Comment (rekt0x) [inline thornode_bot.py:230]: How could it be that there is a non-thor address in there?

File: thornode_bot.py [hunk 3]
```diff
@@ -334,23 +361,29 @@ def healthy(context):
...
-    keyboard = [[
-        InlineKeyboardButton('Add THORNode', callback_data='add_thornode'),
-        InlineKeyboardButton('Show THORNode Stats', callback_data='show_stats')
-    ]]
+    keyboard = [[]]
 
-    # Send message
-    context.bot.send_message(chat_id, 'What do you want to do?', reply_markup=InlineKeyboardMarkup(keyboard))
+    for key in user_data.keys():
+        if "thor" in key:
```
Comment (rekt0x) [inline thornode_bot.py:372]: Same here, how could it be that there is a non-thor address in user data?

File: thornode_bot.py [hunk 4]
```diff
@@ -432,7 +466,7 @@ def error(update, context):
 """
 
 # Conversation state(s)
-WAIT_FOR_USER_INPUT = range(1)
+WAIT_FOR_USER_INPUT, WAIT_FOR_DETAIL = range(2)
```
Comment (rekt0x) [inline thornode_bot.py:469]: Maybe rename those states to be more explanatory:
```
WAIT_FOR_ADDRESS, WAIT_FOR_ACTION = range(2)
```

File: thornode_bot.py (hunk 2 above)
Comment (mr-rooftop) [reply thornode_bot.py:230]: Now a dict for every added thornode address is stored in the user_data dict:
user_data['thorXXX'] = {"status": "active", "bond": "42", "slash_points": "0"}

To get all added addresses, I need to query all user_data keys. 
However, there are also other keys in user_data, like 'block_height_stuck_count' or 'job_started'. 
It's necessary to filter out these non-thor address keys of course.

File: thornode_bot.py (hunk 3 above)
Comment (mr-rooftop) [reply thornode_bot.py:372]: See above. I thought this is a cleaner solution than keeping a dict in a dict in a dict (3 times nested), or Idk some foreign key mapping from one list of addresses to another with the values.

File: thornode_bot.py (hunk 2 above)
Comment (rekt0x) [reply thornode_bot.py:230]: Why are you not creating a sub-dict for that? `user_data['nodes']['thorXXX']`

File: thornode_bot.py (hunk 3 above)
Comment (rekt0x) [reply thornode_bot.py:372]: Also see above, it's pretty common to have nested dicts.

**Pull request merged**